sierra_adapter/sierra_items_to_dynamo/src/test/scala/uk/ac/wellcome/platform/sierra_items_to_dynamo/services/SierraItemsToDynamoWorkerServiceTest.scala

```diff
@@ -33,116 +36,153 @@ class SierraItemsToDynamoWorkerServiceTest
-                    (
-                      sierraItemsToDynamoWorkerService,
-                      queuePair,
-                      table,
-                      metricsSender
-                    ))
+                  actorSystem,
+                  metricsSender) { _ =>
+                  sendNotificationToSQS(queue = queue, message = record2)
+
+                  eventually {
+                    assertStored[SierraItemRecord](
+                      bucket = bucket,
+                      table = table,
+                      id = record1.id.withoutCheckDigit,
+                      record = expectedRecord
+                    )
+                  }
                 }
               }
+            }
           }
         }
       }
     }
   }
 
-  it("reads a sierra record from SQS and inserts it into DynamoDB") {
-    withSierraWorkerService {
-      case (_, QueuePair(queue, _), table, _) =>
-        val bibIds = createSierraBibNumbers(count = 5)
-
-        val bibIds1 = List(bibIds(0), bibIds(1), bibIds(2))
-
-        val itemRecord = createSierraItemRecordWith(
-          modifiedDate = olderDate,
-          bibIds = bibIds1
-        )
-
-        Scanamo.put(dynamoDbClient)(table.name)(itemRecord)
-
-        val bibIds2 = List(bibIds(2), bibIds(3), bibIds(4))
-
-        val record2 = createSierraItemRecordWith(
-          id = itemRecord.id,
-          modifiedDate = newerDate,
-          bibIds = bibIds2
-        )
-
-        sendNotificationToSQS(queue = queue, message = record2)
+  it("returns a GracefulFailureException if it receives an invalid message") {
+    withLocalDynamoDbTable { table =>
+      withLocalS3Bucket { bucket =>
+        withItemRecordVHS(table, bucket) { versionedHybridStore =>
+          withLocalSqsQueueAndDlq {
+            case queuePair @ QueuePair(queue, dlq) =>
+              withActorSystem { actorSystem =>
+                withMockMetricSender { metricsSender =>
+                  withSierraWorkerService(
+                    versionedHybridStore,
+                    queue,
+                    actorSystem,
+                    metricsSender) { _ =>
+                    val body =
+                      """
+                      |{
+                      | "something": "something"
+                      |}
+                    """.stripMargin
+
+                    sendNotificationToSQS(queue = queue, body = body)
+
+                    eventually {
+                      assertQueueEmpty(queue)
+                      assertQueueHasSize(dlq, size = 1)
+                      verify(metricsSender, never()).incrementCount(
+                        "SierraItemsToDynamoWorkerService_ProcessMessage_failure")
+                    }
+                  }
+                }
+              }
+          }
+        }
+      }
+    }
+  }
 
-        val expectedBibIds = List(bibIds(2), bibIds(3), bibIds(4))
-        val expectedUnlinkedBibIds = List(bibIds(0), bibIds(1))
+  def storeSingleRecord(
```

Comment:
Can we put this somewhere where it can be shared? Possibly as a type generic function in `LocalVersionedHybridStore`?

Reply:
I tried that, but it got upset about the implicit HybridRecordEnricher when I passed around a generic VHS instance and I couldn't be bothered to make it work properly!

Reply:
ok fair enough